### PR TITLE
chore: upgrade smithy to 1.13

### PR DESCRIPTION
**Pull request by aajtodd**
## Description of changes

* Upgrade to latest version of smithy
    * Add error flattening transform related to https://github.com/awslabs/smithy/pull/919

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: smithy-kotlin-codegen/build.gradle.kts [hunk 1]
```diff
@@ -42,6 +42,8 @@ val generateSdkRuntimeVersion by tasks.registering {
     // this keeps us from having to manually change version numbers in multiple places
     val resourcesDir = "$buildDir/resources/main/software/amazon/smithy/kotlin/codegen/core"
     val versionFile = file("$resourcesDir/sdk-version.txt")
+    val gradlePropertiesFile = rootProject.file("gradle.properties")
+    inputs.file(gradlePropertiesFile)
```
Comment (kggilmer) [inline smithy-kotlin-codegen/build.gradle.kts:46]: question

This is to determine build dirty-ness or something else?

**Review by kggilmer — APPROVED** [review]
LGTM

File: smithy-kotlin-codegen/build.gradle.kts (hunk 1 above)
Comment (aajtodd) [reply smithy-kotlin-codegen/build.gradle.kts:46]: Yes. When we upgrade the version the generated version is not changing because it is already generated. This generally seems to only affect local development because CI is a fresh build. The fix is to clean and rebuild (or manually delete the file).

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -7,9 +7,4 @@
 <!--- Why is this change required? What problem does it solve? -->
 
 
-## Scope
-<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
-- [ ] Bug fix (non-breaking change which fixes an issue)
-- [ ] New feature (non-breaking change which adds functionality)
-
```
Comment (ianbotsf) [inline .github/pull_request_template.md:14]: **Correctness**: Unrelated to upgrading Smithy. This should be moved to a separate commit.

**Review by ianbotsf — APPROVED** [review]
Minor note about a file that doesn't belong in this commit. Otherwise, looks good.
